edtLHD/src/main/java/fr/univtln/lhd/model/entities/users/Professor.java

```diff
@@ -51,6 +51,12 @@ public int hashCode() {
         return super.hashCode();
     }
 
-    public String getDisplayName() { return getFname().toUpperCase() + ' ' + getName().charAt(0) + '.'; }
-
+    /**
+     * Get a formatted composed of the last name in full caps and first name initials.
+     * @return Formatted string
+     */
+    public String getDisplayName() {
+        return getName().toUpperCase() + ' ' +
+                getFname().replaceAll("([^ -])[^ -]*", "$1.").toUpperCase();
+    }
```

Comment:
This manages the Jean-Marc Robert case 

Was abbreviated `ROBERT J.` instead of `Robert J.-M.`